posts/dedekind-domains-and-class-number-in-lean.md

```diff
@@ -0,0 +1,75 @@
+---
+author: 'Anne Baanen'
+category: ''
+date: 2021-10-01 11:40:57 UTC+02:00
+description: ''
+has_math: true
+link: ''
+slug: dedekind-domains-and-class-number-in-lean
+tags: ''
+title: Dedekind domains and class number in Lean
+type: text
+---
+Pull request [#9701](https://github.com/leanprover-community/mathlib/pull/9071) marks the completion of a string of additions to mathlib centered around formalizing Dedekind domains and the class number of a global field.
+Formalizing these subjects has been one of my long-term goals for mathlib,
+and as far as we are aware, Lean is the first system in which this level of algebraic number theory is available.
+These formalizations have been joint work:
+most notable contributors on the Lean side were Ashvni Narayanan, Filippo Nuccio and myself,
+with Sander Dahmen developing a new finiteness proof of the class group specially for this project.
+Of course, we could not have done this without the assistance of the entire mathlib community.
+[Sander, Ashvni, Filippo and I wrote [a paper](https://github.com/lean-forward/class-number) on the formalization project for the [ITP 2021](http://easyconferences.eu/itp2021/) conference;
+this blog post goes through the highlights of the paper. 
+
+Algebraic number theory is an associative term:
+parsing it as (algebraic (number theory)) we get a subarea of number theory, the study of the (positive) integer numbers, that uses algebraic techniques to solve equations such as $x^2 + 2 = y^3$.
+Alternatively, we can parse ((algebraic number) theory) as the area of mathematics studying [algebraic](https://leanprover-community.github.io/mathlib_docs/ring_theory/algebraic.html#is_algebraic) numbers, those satisfying a polynomial equation $f(\alpha) = 0$ for some nonzero polynomial $f$ with rational coefficients.
+Algebraic numbers are found in [*number fields*](https://leanprover-community.github.io/mathlib_docs/number_theory/number_field.html#number_field), which are finite extensions of the field of rational numbers,
+or equivalently fields generated by adjoining an algebraic element $\alpha$ to $\Q$ (by virtue of the [primitive element theorem](https://leanprover-community.github.io/mathlib_docs/field_theory/primitive_element.html#field.exists_primitive_element)).
+Much like $\Q$ contains the integers $\Z$ as a subring, a number field $K$ contains a [*ring of integers*](https://leanprover-community.github.io/mathlib_docs/number_theory/number_field.html#number_field.ring_of_integers) $O_K$,
```

Comment:
I wonder if we want to use the `find` style links like `docs#` on zulip uses, to prevent these links breaking when they inevitably get moved between files.

Reply:
On the other hand, `find` might not help too much if we rename the defs as well. Perhaps the best solution is some form of link-integrity linter?

Reply:
Ideally we would have monthly archives for the documentation, so that we could link to the version of `number_field` from October 2021 and it will still work in October 2027 when we're done with the mathlib5 port.